database/altsig.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package database
+
+import (
+	"errors"
+	"fmt"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+// ErrMaxAltSigs is returned when attempting to add a ticket's alternate
+// signature but one already in the database.
+var ErrMaxAltSigs = errors.New("max alt sigs")
+
+// The keys used to store the altsig in the database.
+var (
+	altSigAddrK        = []byte("altsig")
+	altSigHistoryAddrK = []byte("altsighistoryaddr")
+	altSigHistoryReqK  = []byte("altsighistoryreq")
+	altSigHistorySigK  = []byte("altsighistorysig")
+)
+
+// AltSigHistory holds the information needed to prove that a client changed
+// their alternate signature address.
+type AltSigHistory struct {
+	// Addr is the address used to validate Sig.
+	Addr string
+	// Req is the original request to set an alternate signature.
+	Req []byte
+	// Sig is the request's signature signed by the private key that
+	// corresponds to the address.
+	Sig []byte
```

Comment:
```suggestion
	ReqSig []byte
```

Lets be specific about this because the name `sig` is used a lot in this code.

Reply:
Ok.

Reply:
9aebd65